### ISSUE TITLE
Conversion of Python Parameters can fail due to rounding issues

### ISSUE THREAD

**Issue by codingS3b** [ComputationalRadiationPhysics/picongpu#2586]
The check for values to be in a certain range fails due to rounding issues. Check out the example below

```python
from picongpu.input.parameters import Parameter
dt = 1.39e-16
tbg_steps = Parameter(
                name="TBG_steps", ptype="run", unit="ps",
                default=1.0, range=(0.1, 10.0),
                pic_to_SI=lambda steps: steps * dt,
                pic_from_SI=lambda time: int(time / dt),
                label="simulation time")

ui_range = tbg_steps.range
print("ui_range: ", ui_range)
pic_range = tbg_steps.convert_to_PIC(list(ui_range))
print("corresponding internal range: ", tuple(pic_range))
print("convert ui range back to pic range: ", tbg_steps.convert_from_PIC(list(pic_range)))
```
raises 
`ValueError: Invalid values found! Values should be contained in self.range `

This could be fixed by not actually using the `range` argument as passed in the constructor but to store
`self.range = tuple(self.convert_from_PIC(self.convert_to_PIC(range)))` internally.
I can provide a PR for this.

Formally more correct would be to store the range on the pic-scale as 'ground truth' values and store them as `self.pic_range` and also do all checks based on this parameter.
The ui scale values are only used for UI purposes and are considered approximate.

This would also be the case for handling discrete values `self.values`  (also for the quick fix)

**Comment by ax3l** [comment]
great catch! your solution sounds legit to me.

**Comment by codingS3b** [comment]
I think if we did not store it, then the example would still fail since:
1. `self.range = (0.1, 10.0)`
2. when calling `convert_to_PIC(vals=list((0.1, 10.0)))`, we check the values via `res = all([self.range[0] <= v <= self.range[1] for v in vals])`. If we replaced this check by the double conversion
```python
allowed_range = tuple(self.convert_from_PIC(self.convert_to_PIC(self.range)))
res = all([allowed_range[0] <= v <= allowed_range[1] for v in vals])
```
we would end up with the same error, since now the `allowed_range = (0.099941, 9.999937999999998)`

**Comment by ax3l** [comment]
cc @kossag14 this is the issue you are seeing

**Comment by ax3l** [comment]
@codingS3b an alternative solution would just be to increase the range on conversion once via the currently used floating-point epsilon: https://stackoverflow.com/questions/9528421/value-for-epsilon-in-python

**Comment by codingS3b** [comment]
Fixed by #2597
